### PR TITLE
276 interface for generating retrofit strategy

### LINKED ISSUE #276: Interface for generating retrofit strategy [frontend]
```[tasklist]
### Tasks
- [ ] https://github.com/IN-CORE/incore-playbook/issues/336
- [ ] https://github.com/IN-CORE/incore-playbook/issues/371
```

---
Design see: 
https://www.figma.com/file/ptmRTl4wUwO3R1guMrb5sO/Community-App-(2024)?type=design&node-id=0%3A1&mode=design&t=2RncEvBJZhU9sTrZ-1

---
Related to issue https://github.com/IN-CORE/dev-scripts/issues/46

- Frontend interface to generate retrofit strategies. A query builder to generate criteria, filters, apply retrofit key and values
E.g. look at QGIS
Image

- Research libraries that we can leverage 
e.g. https://react-querybuilder.js.org/
Image

- Need to consider how to interact with the backend piece to execute the query. Have an endpoint? Have a separate Datawolf flow?

### PR DESCRIPTION
**Pull request by Rashmil-1999**
Ready to be reviewed

~~I have not added anything for step 3 as there are lots of unknowns in what and how we will receive the data. Its better to address them in another task when we have backend ready. Please test it extensively trying to break anything... (consider any dumb moves, it shouldn't break)~~

### CODE REVIEW

**Comment by Rashmil-1999** [comment]
Also, I need fresh SLC values so that I can update the global file

**Comment by Rashmil-1999** [comment]
@longshuicy  This PR is ready to be reviewed.

File: src/globals.js [hunk 1]
```diff
@@ -241,6 +200,171 @@ globals["inputDatasetTypes"] = {
...
+		WestJordanCity: "West Jordan City",
+		SandyCity: "Sandy City",
+		Millcreek: "Millcreek",
+		EmigrationCanyonMetroTownship: "Emigration Canyon Metro Township",
+		MagnaMetroTownship: "Magna Metro Township",
+		CoppertonMetroTownship: "Copperton Metro Township",
+		TownofBrighton: "Town of Brighton",
+		Unincorporated: "Unincorporated",
+		URMM: "URMM",
+		URML: "URML",
```
Comment (longshuicy) [inline src/globals.js]: label-value mapping needed for:

- Archetype for Galveston and Joplin 1,2,3,4,5 --> wood residential building
- Structure type for SLC URMM --> masonary building

File: src/components/childComponents/map/ZoneSelectionMap.jsx [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+				// Directly attempt to populate names and the name-to-feature map
+				const tmpZoneNames = zoneLayerSource
+					.getFeatures()
+					.map((feature) => feature.get("name"));
+				const uniqueSortedZoneNames = [...new Set(tmpZoneNames)].sort((a, b) =>
+					a.localeCompare(b),
+				);
+
+				// Update state with the unique, sorted zone names
+				setZoneNames(uniqueSortedZoneNames);
```
Comment (longshuicy) [inline src/components/childComponents/map/ZoneSelectionMap.jsx]: @longshuicy can dispatch and store in the redux store here

**Pull request merged**